### DISCUSSION TITLE
Temperature sensors

### DISCUSSION THREAD

**Discussion by mikerenna** [gregmac/Genmon-ESP32-Serial-Bridge#7] [Q&A]
Any guidance on where to locate the two temperature sensors inside the generac unit?  

Thanks
Mike

**Comment by gregmac** [comment]
The box I have the ESP32 in is attached with velcro tape to the right side panel right next to the antenna, and the small DS18B20 "ambient" sensor is just hanging down from it, in front of the air intake. 

It very closely tracks with a weather station I have outside, even if the generator is on.

![image](https://user-images.githubusercontent.com/1216375/194202544-db852aba-ec0c-4d7c-a598-581077361219.png)

The other one I went pretty low effort on, it's literally just sitting in an unused mounting hole (?) on some part of the engine near the exhaust.

When the engine is off, it's noticeably lagged behind the ambient temperature. When the engine is running it definitely gets hotter, but I don't know what "normal" temperatures are for it, and don't have any alarms or other automations triggered based on it yet.

![image](https://user-images.githubusercontent.com/1216375/194202799-83624f24-1a6f-4388-a76a-646a806caa4b.png)

**Reply by FlaMike** [reply]
Thank you for posting these pictures, as I was wondering the same thing as the original post.  

I do have one question...  It looks like the ESP and all are outside of the generator case.  I was planning to put everything inside the generators case & just have the antenna external to the generator.  Is there a reason you (may have) mounted the electronics outside of the generator?  (I've not uncovered the part of the generator case by the Evolution controller yet; I've only had the generator a couple of weeks.)  If there isn't room for the ESP, TTL converter, wiring, etc. I'm in the dark since I've not explored deeply enough into the generator case.

**Reply by gregmac** [reply]
Sorry if I was misleading, but mine is mounted on the inside of that side panel, basically underneath the controller (where I drew the yellow box, but inside). 

The whole side area is pretty spacious on my generator: It's the engine air intake, has the battery, and some of the controller wiring. 

I was mainly limited by the length of the antenna cable (5" or so). Also, from that spot the connection to the aux port is only maybe 8" away, but I used about a 24" cable which still lets you take the side panel off and lean it against the back without disconnecting anything.

**Reply by mikerenna** [reply]
Thank you Greg. This was all very helpful. I was pleasantly surprised to realize I had done a similar thing with both probes.  My ambient probe is usually just a degree or so warmer than what the weather services tells me the outside temperature is, but I still like having an idea of what the temperature inside that section of the house is. I actually was thinking about repositioning it so it's less indicative of outside temp and more representative of that space where the ESP encloser lives.  My engine probe isn't touching the engine but I wonder if maybe that's actually a better idea. I suspect it'll heat up more but also be more indicative of the engine temp as a whole? I think I'll play around with. Overall this is a terrific alternative to the raspberry pi solution. Thanks for sharing your idea.

**Reply by FlaMike** [reply]
No worries at all, Greg.  I just wanted to confirm the location of the ESP and all before diving in.  The short antenna cables are always a challenge.  As long as the ESP can be mounted nearby as you did, it's not a significant limitation.

Extending the ATX cable so it's long enough to remove the side panel without disconnecting anything is a great idea, too.  I plan to do that as well.  

Any & all other tips you have are appreciated greatly!

The ESP is coming next week & I already have everything else, so I'm ready to roll up my sleeves & get going on the installation.

Thanks again for making your solution available & your guidance along the way!
